Src/FluentAssertions/Formatting/DefaultValueFormatter.cs

```diff
@@ -62,6 +63,13 @@ protected virtual MemberInfo[] GetMembers(Type type)
         return type.GetNonPrivateMembers(MemberVisibility.Public).ToArray();
     }
 
+    private static bool HasCompilerGeneratedToStringImplementation(object value)
+    {
+        Type type = value.GetType();
+
+        return HasDefaultToStringImplementation(value) || type.IsAnonymousType() || type.IsRecord();
```

Comment:
❓ What about tuples? Are they formatted correctly?
🔧 I think this entire `HasCompilerGeneratedToStringImplementation` should be moved to `TypeExtensions`, including a cache.

Reply:
Good point about tuples, I've added a check for them as well.

`HasCompilerGeneratedToStringImplementation` relies in part on the value of a particular instance.  I've moved `IsCompilerGeneratedType` into `TypeExtensions` with a cache.